inginious/frontend/plugins/user_management/update_user_data.py

```diff
@@ -0,0 +1,214 @@
+                                                             collection_manager)
+    return count
+
+
+def close_user_sessions(username, collection_manager):
+    """ Close the open sessions of a user """
+    update_filter = {
+        "data.username": username
+    }
+    close_sessions = {
+        "$set": {
+            "data.loggedin": False
+        }
+    }
+
+    return collection_manager.update_many_in_collection("sessions", update_filter, close_sessions)
+
+
+def add_block_user(username, collection_manager):
+    """ Add the field block to a user's document in users collection to indicate that the user can't do logging"""
+    user_filter = {
+        "username": username
+    }
+    new_name = {
+        "$set": {"block": True}
+    }
+
+    return collection_manager.update_collection("users", user_filter, new_name)
+
+
+def unlock_user(username, collection_manager):
+    """ Removes the block field from the user's document """
+    user_filter = {
+        "username": username
+    }
+    new_name = {
+        "$unset": {"block": ""}
+    }
+
+    return collection_manager.update_collection("users", user_filter, new_name)
+
+
+def is_array(collection_information):
+    """ return a boolean value to determinate if the field that is contents the username is an array """
+    path = collection_information["path"]
+    index_list = sorted(collection_information["index_array"])
+    if not index_list:
+        return False
+
+    num_fields = len(_get_field_end_index_in_path(path))
+    last_index_path = num_fields - 1
+    return last_index_path == index_list[-1]
+
+
+def _crete_update_to_change_username_collection(username, new_username, coll_name, collection_info, collection_manager):
+    """ For each sub set of information about a collection (path and index_array), make a update process
+    and count the number of changes.
+     """
+    count = 0
+    for info in collection_info:
+        ans = _create_update_to_change_username(username, new_username, coll_name, info, collection_manager)
+        count += ans
+    return count
+
+
+def _create_update_to_change_username(username, new_username, coll_name, collection_info_field, collection_manager):
+    """ update the username with the new username in a specific path """
+    user_filter = _create_user_filter(username, collection_info_field["path"])
+    path = _process_path(collection_info_field)
+    there_are_changes = True
+    change_username_count = 0
+    # positional operator "$" only acts as a placeholder for the first element that matches the query document
+    # therefore is necessary (in mongodb 3.4) repeat the request to get to all objects in an array of objects
+    if is_array(collection_info_field):
+        push_new_username = _create_push_operator(path, new_username)
+        pull_old_username = _create_pull_operator(path, username)
+        while there_are_changes:
```

Comment:
Because positional operator `$` only acts as a placeholder for the first element that matches the query document, therefore, is necessary (in mongodb 3.4) repeat the request to access all objects objects in an array of objects until there are no more changes